src/tribler-core/tribler_core/components/simple_storage/simple_storage.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import logging
+
+from pydantic import BaseModel
+
+from tribler_core.utilities.path_util import Path
+
+DEFAULT_SAVE_INTERVAL = 5 * 60  # force Storage to save data every 5 minutes
+
+
+class StorageData(BaseModel):
+    last_processed_torrent_id: int = 0
+
+
+class SimpleStorage:
+    """ SimpleStorage is object storage that stores data in JSON format and uses
+    `pydantic` `BaseModel` for defining models.
+
+    It stores data on a shutdown and every 5 minutes.
+    """
+    def __init__(self, path: Path, save_interval: float = DEFAULT_SAVE_INTERVAL):
+        """
+        Args:
+            path: path to the file with storage. Could be a path to a non existent file.
+            save_interval: interval in seconds in which the storage will store a data to a disk.
+        """
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.data = StorageData()
+
+        self.path = path
+        self.save_interval = save_interval
+
+        self._loop = asyncio.get_event_loop()
+        self._task: asyncio.TimerHandle = self._loop.call_later(self.save_interval, self._save_and_schedule_next)
+
+    def load(self):
+        """ Load data from `self.path`. In case the file doesn't exists, the function
+        will create the data with defaults values.
+        """
+        self.logger.info(f'Loading storage from {self.path}')
+        try:
+            self.data = StorageData.parse_file(self.path)
+        except FileNotFoundError:
```

Comment:
Just to initialize storage with default values?

Reply:
Eg:
```python
    def load(self):
        """ Load data from `self.path`. In case the file doesn't exist, the function
        will create the data with defaults values.
        """
        self.logger.info(f'Loading storage from {self.path}')
        try:
            self.data = StorageData.parse_file(self.path)
        except OSError:
            self.logger.info('The storage file does not exist or corrupted. Create a new storage.')
            self.data = StorageData()
        self.logger.info(f'Loaded storage: {self.data}')
```

Reply:
Yes, exactly

Reply:
I think, `pydantic.ValidationError` should be handled as well, for the case of corrupted JSON

Reply:
No, I've checked, in the case of a corrupted JSON the following exception was raised `json.decoder.JSONDecodeError`, so I decided to add a broad exception handling for this block.